### PR TITLE
Add XYZ tiles to Datasource manager dialog and Layers menu

### PR DESCRIPTION
**Pull request by alexbruy**
## Description
Adds new tab to Datasource Manager dialog allowing to manage/add XYZ connections. Also add corresponding entry to Layers menu.

### CODE REVIEW

File: src/providers/wms/qgsxyzsourceselect.cpp [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+    QgsXyzConnectionUtils::addConnection( nc->connection() );
+    populateConnectionList();
+    emit connectionsChanged();
+  }
+  delete nc;
+}
+
+void QgsXyzSourceSelect::btnEdit_clicked()
+{
+  QgsXyzConnectionDialog *nc = new QgsXyzConnectionDialog( this );
```
Comment (elpaso) [inline src/providers/wms/qgsxyzsourceselect.cpp:59]: std::unique_ptr

Comment (elpaso) [reply src/providers/wms/qgsxyzsourceselect.cpp:59]: .. or just allocate on stack

**Review by elpaso — APPROVED** [review]
Long awaited, thank you!

File: src/providers/wms/qgsxyzsourceselect.cpp (hunk 1 above)
Comment (alexbruy) [reply src/providers/wms/qgsxyzsourceselect.cpp]: Fixed.

File: src/gui/qgisinterface.h [hunk 2]
```diff
@@ -479,6 +479,8 @@ class GUI_EXPORT QgisInterface : public QObject
     virtual QAction *actionAddRasterLayer() = 0;
     virtual QAction *actionAddPgLayer() = 0;
     virtual QAction *actionAddWmsLayer() = 0;
+    //! Returns the native Add XYZ layer action.
```
Comment (nyalldawson) [inline src/gui/qgisinterface.h:482]: \since QGIS 3.14

**Review by nyalldawson — COMMENTED** [review]
Nice!!

File: src/gui/qgisinterface.h (hunk 2 above)
Comment (alexbruy) [reply src/gui/qgisinterface.h]: Fixed

**Pull request merged**